packages/common/src/dynamicLoading/dynamic-loading-setting.ts

```diff
@@ -0,0 +1,73 @@
+import type { FontSource } from './types';
+
+declare global {
+  interface ExpoGlobal {
+    modules: {
+      ExpoAsset: {
+        downloadAsync: (uri: string, path: string, type: string) => Promise<string>;
+      };
+      ExpoFontLoader: {
+        loadAsync: (fontFamilyAlias: string, fileUri: string) => Promise<void>;
+        getLoadedFonts: () => string[];
+      };
+    };
+  }
+
+  // eslint-disable-next-line vars-on-top
+  var expo: ExpoGlobal | undefined;
+}
+
+const hasNecessaryExpoModules = !!globalThis.expo?.modules?.ExpoAsset && !!globalThis.expo?.modules?.ExpoFontLoader;
+
+const hasNecessaryExpoFeatures =
+  hasNecessaryExpoModules && typeof globalThis.expo?.modules.ExpoFontLoader.getLoadedFonts === 'function';
+
+let dynamicFontLoadingEnabled = hasNecessaryExpoFeatures;
+
+export const isDynamicLoadingSupported = () => hasNecessaryExpoFeatures;
+
+/**
+ * Set whether dynamic loading of fonts is enabled.
+ * Currently, the presence of Expo Asset and Font Loader modules is a prerequisite for enabling.
+ * In the future, React Native core apis will be used for dynamic font loading.
+ *
+ * @param value - whether dynamic loading of fonts is enabled
+ * @returns `true` if dynamic loading of fonts was successfully set. `false` otherwise.
+ * */
+export const setDynamicLoadingEnabled = (value: boolean): boolean => {
+  dynamicFontLoadingEnabled = !!value;
+  if (!hasNecessaryExpoFeatures) {
```

Comment:
Should the if run first and return early?

Reply:
indeed, that's better. Did that.